dd-sdk-android-realm/src/main/kotlin/com/datadog/android/realm/DatadogRealmExt.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.realm
+
+import com.datadog.android.rum.GlobalRum
+import com.datadog.android.rum.RumAttributes
+import com.datadog.android.rum.RumErrorSource
+import io.realm.Realm
+import java.io.Closeable
+internal const val REALM_ERROR_MESSAGE = "Error while executing Realm transaction"
+
+/**
+ * Executes the given [block] function on this [Realm] instance
+ * and then closes it down correctly whether an exception
+ * is thrown or not.
+ * In case the [block] will throw any exception this will be intercepted and propagated as
+ * a Rum error event.
+ * @param block a function to process this [Closeable] resource.
+ * @return the result of [block] function invoked on this resource.
+ */
+@Suppress("TooGenericExceptionCaught")
+fun <R> Realm.useWithRum(block: (Realm) -> R): R {
```

Comment:
As I said in daily, I don't really see much value into creating an extension and artefact just for this few lines of code. I'd rather see this extension in our `ktx` module for `Closable` types. 
As for the name, i think something like `useMonitored` might be a better name. We could even have a `useTraced` extension too that allow users to create a span around a closable ;)

Reply:
Ok, I will move this extensions in the Closable and make it generic.